src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -83,6 +83,23 @@ public TestCase ConvertTestCase(XmlNode testNode)
             return testCase;
         }
 
+        /// <summary>
+        /// Knows how to convert an entire XML fragment.
+        /// </summary>
+        internal IList<TestCase> ConvertTestCases(string xml)
```

Comment:
You appear to have added a method to the adapter, which is only their for tests to use. Am I missing something?

Reply:
Yes. This was in hopes that it would be canonicalized and that the adapter might end up using it.

Reply:
If there are three places that require you to know to use `.SelectNodes("//test-case")` in order to use `ConvertTestCases`, what happens when there's a subtle change in one of the three? (An example, I wondered about was `.SelectNodes("test-case")`?)

Reply:
I did. This was in hopes that it would be canonicalized and that the adapter might end up using it. If there are three places that require you to know to use `.SelectNodes("//test-case")` in order to use `ConvertTestCase`, what happens when there's a subtle change in one of the three? (An example, I wondered about `.SelectNodes("test-case")`?) I was hoping that would potentially end up encapsulated here.

Reply:
But your test is testing a method that is not used in the adapter itself... right now anyway. So you are not fully testing the application as it exists.

A refactoring of the code is another matter, but we should have a reason to undertake that refactoring and we should do it separately from adding a feature. All IMO, of course.

Searching the code online, I only found two places in the adapter where the converter is called, plus a bunch in the tests. I could have missed one. The two calls I found are in the Discoverer and the Executor. It could be factored into a method of the base adapter class.

Reply:
The test usage made three, and I didn't look for others.

> So you are not fully testing the application as it exists.

I'm missing something here. Since it purely delegates to the method I would be calling anyway, I'm covering no less than I would be covering without having added this method. Right?

I'll go ahead and move this into the test layer since it's too speculative.